src/sources/syslog_remap.rs

```diff
@@ -0,0 +1,97 @@
+use crate::sources::socket::SocketConfig;
+use crate::{
+    config::{
+        log_schema, DataType, GenerateConfig, Resource, SourceConfig, SourceContext,
+        SourceDescription,
+    },
+    transforms::remap::{Remap, RemapConfig},
+    Pipeline,
+};
+use bytes::Bytes;
+use indoc::indoc;
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct SyslogRemapConfig {
+    // Config settings we may need access to
+    // host_key: Option<String>,
+    #[serde(flatten)]
+    original_config: SocketConfig,
+}
+
+inventory::submit! {
+    SourceDescription::new::<SyslogRemapConfig>("syslog_remap")
+}
+
+impl GenerateConfig for SyslogRemapConfig {
+    fn generate_config() -> toml::Value {
+        SocketConfig::generate_config()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "syslog_remap")]
+impl SourceConfig for SyslogRemapConfig {
+    async fn build(&self, mut cx: SourceContext) -> crate::Result<super::Source> {
+        let conf = RemapConfig {
+            source: indoc! {r#"
+                structured = parse_syslog!(.message)
+                . = merge(., structured)
+            "#}
+            .to_string(),
+            drop_on_abort: false,
+            drop_on_error: false,
+        };
+        let tf = Remap::new(conf).unwrap();
+        let (to_transform, rx) = Pipeline::new_with_buffer(100, vec![Box::new(tf)]);
+        let out = cx.out;
+        cx.out = to_transform;
+        tokio::spawn(async move {
```

Comment:
Instead of creating a new pipeline and spawning an additional task, I think we can just do something like:

```rust
cx.out.inlines.push(Box::new(tf));
```

We may want a nicer API than just making `inlines` a public field, but the idea is the same.